src/microbe_stage/MembranePointData.cs

```diff
@@ -36,6 +38,9 @@ public sealed class MembranePointData : IMembraneDataSource, ICacheableData
             }
         });
 
+        finalEngulfMesh = new Lazy<(ArrayMesh Mesh, int SurfaceIndex)>(() =>
+            MembraneShapeGenerator.GetThreadSpecificGenerator().GenerateEngulfMesh(this));
```

Comment:
This needs to be rewritten to lock similarly to how it is done on line 35 due to that linked issue https://github.com/Revolutionary-Games/Thrive/issues/4989